geometry.go

```diff
@@ -49,6 +49,21 @@ func (u Vec) XY() (x, y float64) {
 	return u.X, u.Y
 }
 
+// u->v, rotated by math.pi/2
+func (u Vec) Normal(v Vec) Vec {
+	return Vec{X: u.Y - v.Y, Y: v.X - u.X}
+}
+
+// u->v, rotated by math.pi/2, scaled to length
+func (u Vec) ScaledNormal(v Vec, length float64) Vec {
+	u = Vec{X: u.Y - v.Y, Y: v.X - u.X}
+	l := math.Sqrt(u.X*u.X + u.Y*u.Y)
+	if l > 0 {
+		length /= l
+	}
+	return Vec{X: length * u.X, Y: length * u.Y}
+}
+
```

Comment:
I dislike this function. Duplicates code, this is something we should leave to the inlining capabilities of the compiler (which improve over time).